src/build/genprod.scala

```diff
@@ -353,7 +352,7 @@ object ProductTwo extends Product(2)
 
 class Product(val i: Int) extends Group("Product") with Arity {
   val productElementComment = """
-  /** Returns the n-th projection of this product if 0 < n <= productArity,
+  /** Returns the n-th projection of this product if 0 <= n < productArity,
    *  otherwise throws an `IndexOutOfBoundsException`.
    *
    *  @param n number of the projection to be returned
```

Comment:
This version of the main comment avoids having to decide if projection indexing starts at 0 or 1. It is also consistent with the commenting on the `_n` defs,

    Returns the projection of element n + 1 of this product if 0 <= n < productArity,

Technically accurate but possibly less clear. This change could just be left as-is and would be an improvement.